sdk/ml/azure-ai-ml/tests/dsl/unittests/test_dsl_pipeline.py

```diff
@@ -1940,3 +1940,53 @@ def pipeline_func(component_in_path):
         pipeline_job = pipeline_func(component_in_path=Data(name="test", version="1", type=AssetTypes.MLTABLE))
         result = pipeline_job._validate()
         assert result._to_dict() == {"result": "Succeeded"}
+
+    def test_pipeline_with_non_pipeline_parameters(self):
+        component_yaml = components_dir / "helloworld_component.yml"
+        component_func1 = load_component(source=component_yaml, params_override=[{"name": "component_name_1"}])
+        component_func2 = load_component(source=component_yaml, params_override=[{"name": "component_name_2"}])
+
+        @dsl.pipeline(non_pipeline_parameters=["other_params"])
+        def pipeline_func(job_in_number, job_in_path, other_params):
+            component_func1(component_in_number=job_in_number, component_in_path=job_in_path)
```

Comment:
let's use the non_pipeline_paramters in the dsl.pipeline body

Reply:
Add test case to use non_pipelne_parameters in dsl.pipeline body